guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/upload/UploadFormViewImpl.java

```diff
@@ -0,0 +1,201 @@
+                    hide();
+                }
+            });
+        }}));
+        this.setTitle(M2RepoEditorConstants.INSTANCE.ArtifactUpload());
+        this.add(doUploadForm());
+    }
+
+    @Override
+    public void hide() {
+        hideGAVInputs();
+        super.hide();
+    }
+
+    public WellForm doUploadForm() {
+        form = new WellForm();
+        form.setAction( getWebContext() + "/maven2wb" );
+        form.setEncoding( FormPanel.ENCODING_MULTIPART );
+        form.setMethod( FormPanel.METHOD_POST );
+
+        form.addSubmitHandler(new Form.SubmitHandler() {
+            @Override
+            public void onSubmit(final Form.SubmitEvent event) {
+                presenter.handleSubmit(event);
+            }
+        });
+
+        uploader = new FileUpload(new Command() {
+            @Override
+            public void execute() {
+                form.submit();
+            }
+        });
+
+        uploader.setName(HTMLFileManagerFields.UPLOAD_FIELD_NAME_ATTACH);
+
+        form.addSubmitCompleteHandler(new Form.SubmitCompleteHandler() {
+            public void onSubmitComplete(final Form.SubmitCompleteEvent event) {
+                presenter.handleSubmitComplete(event);
+            }
+        });
+
+        HorizontalPanel fields = new HorizontalPanel();
+        fields.add(uploader);
+
+        hiddenGroupIdField.setName( HTMLFileManagerFields.GROUP_ID );
+        hiddenArtifactIdField.setName( HTMLFileManagerFields.ARTIFACT_ID );
+        hiddenVersionIdField.setName( HTMLFileManagerFields.VERSION_ID );
+        hideGAVInputs();
+
+        hiddenFieldsPanel.addAttribute( "GroupID:", hiddenGroupIdField );
+        hiddenFieldsPanel.addAttribute( "ArtifactID:", hiddenArtifactIdField );
+        hiddenFieldsPanel.addAttribute( "VersionID:", hiddenVersionIdField );
+
+        VerticalPanel allFields = new VerticalPanel();
+        allFields.add( fields );
+        allFields.add( hiddenFieldsPanel );
+
+        form.add( allFields );
+
+        return form;
+    }
+
+    private String getWebContext() {
+        String context = GWT.getModuleBaseURL().replace( GWT.getModuleName() + "/", "" );
+        if ( context.endsWith( "/" ) ) {
+            context = context.substring( 0, context.length() - 1 );
+        }
+        return context;
+    }
+
+    public void showUploadingBusy() {
+        BusyPopup.showMessage(M2RepoEditorConstants.INSTANCE.Uploading());
+    }
+
+    public void hideUploadingBusy() {
+        BusyPopup.close();
+    }
+
+    @Override
+    public void showWarning(String message) {
+        Window.alert(message);
+    }
+
+    @Override
+    public void showSelectFileUploadWarning() {
+        showWarning(M2RepoEditorConstants.INSTANCE.SelectFileUpload());
+    }
+
+    @Override
+    public void showUnsupportedFileTypeWarning() {
+        showWarning(M2RepoEditorConstants.INSTANCE.UnsupportedFileType());
+    }
+
+    @Override
+    public void showUploadedSuccessfullyWarning() {
```

Comment:
Being picky, but is "successful upload" really a "warning"?

Would it be better to change the ```showWarning(String message)``` to be ```showMessage(...)```?